ovs/extensions/generic/tests/sshclient_mock.py

```diff
@@ -0,0 +1,419 @@
+                raise ValueError('In unittest mode, the paths must be absolute')
+
+        for file_location in filenames:
+            parts = [part for part in file_location.strip('/').split('/') if part]
+            pointer = MockedSSHClient._file_system[client.ip]
+            for index, part in enumerate(parts):
+                regex = None if '*' not in part else re.compile('^{0}$'.format(part.replace('.', '\.').replace('*', '.*')))
+                if index == len(parts) - 1:
+                    pointer = pointer['files']
+                    if regex is not None:
+                        for sub_file in copy.deepcopy(pointer):
+                            if regex.match(sub_file):
+                                pointer.pop(sub_file)
+                    elif part in pointer:
+                        pointer.pop(part)
+                    break
+
+                pointer = pointer['dirs']
+                if regex is not None:
+                    for sub_dir in pointer:
+                        if regex.match(sub_dir):
+                            MockedSSHClient.file_delete(client=client, filenames='/{0}/{1}/{2}'.format('/'.join(parts[:index]), sub_dir, '/'.join(parts[-(len(parts) - index - 1):])))
+                if part not in pointer:
+                    break
+                pointer = pointer[part]
+
+    @staticmethod
+    def file_unlink(client, path):
+        """
+        Mocked file_unlink method
+        """
+        _ = client, path
+
+    @staticmethod
+    def file_read_link(client, path):
+        """
+        Mocked file_read_link method
+        """
+        _ = client, path
+
+    @staticmethod
+    def file_read(client, filename):
+        """
+        Mocked file_read method
+        """
+        first_part, last_part = MockedSSHClient._split_last_part_from(filename)
+        pointer = MockedSSHClient.traverse_file_system(client=client,
+                                                       path=first_part)
+        if pointer is None or last_part not in pointer['files']:
+            raise OSError("No such file or directory: '{0}'".format(filename))
+
+        contents = pointer['files'][last_part]['contents']
+        try:
+            return json.loads(contents)
+        except ValueError:
+            return contents
+
+    @staticmethod
+    def file_write(client, filename, contents):
+        """
+        Mocked file_write method
+        """
+        if client.ip not in MockedSSHClient._file_system:
+            MockedSSHClient._file_system[client.ip] = {'info': {}, 'dirs': {}, 'files': {}}
+
+        if not filename.startswith('/'):
+            raise ValueError('In unittest mode, the paths must be absolute')
+
+        parts = [part for part in filename.strip('/').split('/') if part]
+        pointer = MockedSSHClient._file_system[client.ip]
+        if isinstance(contents, list) or isinstance(contents, dict):
+            contents = json.dumps(contents, indent=4, sort_keys=True)
```

Comment:
I'd implement the default behavior. I guess internally it will just to `str(...)` or maybe raise.